cmd/cluster/support/post.go

```diff
@@ -83,8 +117,37 @@ func validateResolutionString(res string) error {
 	return nil
 }
 
+func (p *Post) check() error {
+	if p.Template != "" {
+		if p.Problem != "" || p.Resolution != "" || p.Misconfiguration != "" {
+			return fmt.Errorf("\nIf Template flag is present, all three --problem, --resolution and --misconfiguration flags cannot be used")
+		}
+		p.parseUserParameters() // parse all the '-p' user flags
+
+		// For every '-p' flag, replace its related placeholder in the template
+		for k := range userParameterNames {
+			p.replaceFlags(userParameterNames[k], userParameterValues[k])
```

Comment:
We need to parse after reading the template
Ref -https://github.com/openshift/osdctl/blob/2bd6c403d30ce1a06c2c24e30a5b54f3098a9184/cmd/servicelog/post.go#L139